catboost/python-package/catboost/core.py

```diff
@@ -519,13 +519,28 @@ def __init__(
             If -1, then the number of threads is set to the number of CPU cores.
 
         """
+
+        if isinstance(column_description, ColumnDescription):
+            column_type_to_indices, names = column_description.get_metadata()
+
+            cat_features = column_type_to_indices["Categ"]
+            text_features = column_type_to_indices["Text"]
+            embedding_features = column_type_to_indices[""]
+            weight = column_type_to_indices["Weight"]
+            group_id = column_type_to_indices["GroupId"]
+            group_weight = column_type_to_indices["GroupWeight"]
+            subgroup_id = column_type_to_indices["SubgroupId"]
+            baseline = column_type_to_indices["Baseline"]
+            feature_names = names
+            column_description = None
+
```

Comment:
we need to extract `label` also depends on `data` type. When `data` is file then this method should be used for read https://github.com/catboost/catboost/blob/master/catboost/libs/data/load_data.cpp#L128

Reply:
Too hard for me. This method never used before. I tried to add this method in _catboost.pyx but I don't understand where and how I should call him

Reply:
Added a class member that stores the path to which this ColumnDescription is written and in the Pool is simply written to a file and transferred further as a path to file